File: z01_miscellaneous/Scratch Project/Android/ConvertMilesToKM/ConvertMilesToKM2/app/src/main/java/com/example/convertmilestokm/MainActivity.java
package com.example.convertmilestokm;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button buttonConvertmtokm = (Button) findViewById(R.id.buttonConvertmtokm);
        buttonConvertmtokm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText textboxMiles = (EditText) findViewById(R.id.editTextMiles);
                EditText textboxKm = (EditText) findViewById(R.id.editTextKilometers);
                double vMiles = Double.valueOf(textboxMiles.getText().toString());
                double vKm = vMiles / 0.62137;
                DecimalFormat formatVal = new DecimalFormat("##.##");
                textboxKm.setText(formatVal.format(vKm));
            }
        });
        Button buttonConvertkmtom = (Button) findViewById(R.id.buttonConvertkmtom);
        buttonConvertkmtom.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText textboxKm = (EditText) findViewById(R.id.editTextKilometers);
                EditText textboxMiles = (EditText) findViewById(R.id.editTextMiles);
                double vKm = Double.valueOf(textboxKm.getText().toString());
                double vMiles = vKm * 0.62137;
                DecimalFormat formatVal = new DecimalFormat("##.##");
                textboxMiles.setText(formatVal.format(vMiles));
            }
        });
    }
}

File: helloKoding - databinding/app/src/main/java/com/example/hellokoding/MainActivity.kt
package com.example.hellokoding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import kotlinx.android.synthetic.main.activity_main.*
import com.example.hellokoding.databinding.ActivityMainBindingImpl

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBindingImpl

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

//        val button : Button = findViewById(R.id.btn1)

        btn1.setOnClickListener {
//            Toast.makeText(this@MainActivity, "tombol berhasil diklik", Toast.LENGTH_SHORT).show()

            val name = nameText.text.toString()
            textBase.setText(name)

            if (name == null || name.trim() == "")
                Toast.makeText(this@MainActivity, "nama tidak boleh kosong", Toast.LENGTH_SHORT).show()
            else {
                textBase.setText(name)

                //change image
                val imgRes = resources.getIdentifier(name, "drawable", packageName)
                img1.setImageResource(imgRes)
            }
        }
    }
}
